### ISSUE TITLE
Discrepancies in docs, and failure to refresh instances.

### ISSUE THREAD

**Issue by davidkarlsen** [codecentric/spring-boot-admin#1788]
config states to define `spring.cloud.kubernetes.discovery.service-labels` for filtering, but it seems to actually be `spring.boot.admin.discovery.instances-metadata` [1].

Also, sb-admin seems to only discover instances on boot-up, if I scale a deployment up, it won't list the new instances. My config is:

```
package com.tietoevry.fss.baas;

import de.codecentric.boot.admin.server.config.EnableAdminServer;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;
import org.springframework.scheduling.annotation.EnableScheduling;

@SpringBootApplication
@EnableAdminServer
@EnableDiscoveryClient
@EnableScheduling
public class AdminApplication {

  public static void main(String[] args) {
    SpringApplication.run(AdminApplication.class, args);
  }


}
```

[1]
```
application.yaml: |
    spring:
      cloud:
        kubernetes:
          reload:
            enabled: true
          discovery:
            all-namespaces: true
            service-labels:
              spring-boot: "true" <-- no use
      boot:
        admin:
          ui:
            public-url: https://eos-tac-spring-boot-admin-d1.apps.os-global.finods.com
          discovery:
            #services:
            #- "spring-boot*"
            instances-metadata:
              spring-boot: "true" <-- actually need to be set here
```

**Comment by davidkarlsen** [comment]
Maybe related to https://github.com/spring-cloud/spring-cloud-kubernetes/issues/745

**Comment by davidkarlsen** [comment]
ait, got to the bottom of this, it boils down to swithing client impls, as the fabric one can do events and refreshes:
```
 <!-- broken refresh, https://github.com/spring-cloud/spring-cloud-kubernetes/issues/745 
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-starter-kubernetes-client-all</artifactId>
      <version>2.0.3</version>
    </dependency>
    -->
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-kubernetes-fabric8-discovery</artifactId>
      <version>2.0.3</version>
    </dependency>
```

**Comment by SteKoe** [comment]
Hi @davidkarlsen,

thanks for opening this issue. The docs of SBA at chapter [4.3.3](https://codecentric.github.io/spring-boot-admin/2.4.3/#_converting_serviceinstances) at Table 4 states to use `spring.boot.admin.discovery.instances-metadata` to filter instances based on their metadata. I do not see any discrepancies in the docs here, what part do you refer to?

**Comment by davidkarlsen** [comment]
Sorry, ignore me - I mixed the two different manuals.
But maybe emphasis could be put into the docs that the fabric8 variant of discovery is needed for proper refresh?

BTW: I have packaged a docker-image + helm chart in order to install this smoothly on Kubernetes: https://artifacthub.io/packages/helm/evryfs-oss/spring-boot-admin - maybe you want to add a note to your docs in order to increase discoverability?
